libcloud/dns/drivers/worldwidedns.py

```diff
@@ -0,0 +1,506 @@
+        self.connection.request(action, params=params)
+        zone = self.get_zone(domain)
+        if ttl is not None:
+            zone = self.update_zone(zone, zone.domain, ttl=ttl)
+        return zone
+
+    def create_record(self, name, zone, type, data, entry):
+        """
+        Create a new record.
+
+        :param name: Record name without the domain name (e.g. www).
+                     Note: If you want to create a record for a base domain
+                     name, you should specify empty string ('') for this
+                     argument.
+        :type  name: ``str``
+
+        :param zone: Zone where the requested record is created.
+        :type  zone: :class:`Zone`
+
+        :param type: DNS record type (MX, CNAME, A, NS, SRV, TXT).
+        :type  type: :class:`RecordType`
+
+        :param data: Data for the record (depends on the record type).
+        :type  data: ``str``
+
+        :param entry: Entry position (1 thru 40)
+        :type  entry: ``int``
+
+        :rtype: :class:`Record`
+        """
+        if name == '':
+            name = '@'
+        if type not in self.RECORD_TYPE_MAP:
+            raise RecordError(value="Record type is not allowed",
+                              driver=zone.driver,
+                              record_id=name)
+        extra = {'S%s' % entry: name,
+                 'T%s' % entry: type,
+                 'D%s' % entry: data}
+        zone = self.update_zone(zone, zone.domain, extra=extra)
+        record = self.get_record(zone.id, name)
+        return record
+
+    def delete_zone(self, zone):
+        """
+        Delete a zone.
+
+        Note: This will delete all the records belonging to this zone.
+
+        :param zone: Zone to delete.
+        :type  zone: :class:`Zone`
+
+        :rtype: ``bool``
+
+        For more information, please see
+        https://www.worldwidedns.net/dns_api_protocol_delete_domain.asp
+        or
+        https://www.worldwidedns.net/dns_api_protocol_delete_domain_reseller.asp
+        """
+        params = {'DOMAIN': zone.domain}
+        action = '/api_dns_delete_domain.asp'
+        if self.reseller_id is not None:
+            action = '/api_dns_delete_domain_reseller.asp'
+        response = self.connection.request(action, params=params)
+        return response.success()
+
+    def delete_record(self, record):
+        """
+        Delete a record.
+
+        :param record: Record to delete.
+        :type  record: :class:`Record`
+
+        :rtype: ``bool``
+        """
+        zone = record.zone
+        for index in range(MAX_RECORD_ENTRIES):
+            if record.name == zone.extra['S%s' % (index + 1)]:
+                entry = index + 1
+                break
+        extra = {'S%s' % entry: '',
+                 'T%s' % entry: 'NONE',
+                 'D%s' % entry: ''}
+        self.update_zone(zone, zone.domain, extra=extra)
+        return True
+
+    def view_zone(self, domain, name_server):
```

Comment:
This is an extension method so please prefix the method name with `ex_` - `def ex_view_zone(self, domain, name_server):`